website/content/docs/secrets/databases/custom.mdx

```diff
@@ -200,7 +232,37 @@ $ vault write database/config/mydatabase \
     myplugins_connection_details="..."
 ```
 
-## Upgrading database plugins
+## Upgrading database plugins to leverage plugin multiplexing
+
+### Background
+
+Scaling many external plugins can become resource intensive. To address
+performance problems with scaling external plugins, database plugins can be
+made to implement [plugin multiplexing](/docs/plugins/plugin-architecture#plugin-multiplexing)
+which allows a single plugin process to be used for multiple database
+connections. To enable multiplexing, the plugin must be compiled with the
+`ServeMultiplex` function call from Vault's `dbplugin` package.
+
+### Upgrading your database plugin to leverage plugin multiplexing
+
+There is only one step required to upgrade from a non-multiplexed to a
+multiplexed database plugin:
+
+1. Change the `Serve` function call to `ServeMultiplex`.
+
+This will run the RPC server for the plugin just as before. However, the
+`ServeMultiplex` function takes the factory function directly as its argument.
+This factory function is a function that returns an object that implements the
+`dbplugin.Database` interface.
+
+### When should plugin multiplexing be avoided?
+
+Some use cases that should avoid plugin multiplexing might include:
+
+* Plugin process level separation is required
+* The plugin shares state or a global cache which may leak data between each database config
```

Comment:
I don't think this is quite true since the plugin will instantiate a separate `logical.Backend` object on the plugin side, so even things like caches would be okay.

Globally shared resources such as files or environment variables are a bit more problematic, but this issue exists regardless of whether multiplexing is supported or not.

An additional use case is wanting to avoid restart across all mounts/DB connections for a plugin type on crashes or plugin reload calls.